Sources/SwiftlyCore/HTTPClient.swift

```diff
@@ -186,9 +185,5 @@ public struct SwiftlyHTTPClient {
 }
 
 private class HTTPClientWrapper {
```

Comment:
If we go the shared route, I don't think we need this class at all actually. The original intent was to be able to share a single HTTP client via `class`'s arc semantics and also to ensure it shut down when the application was shutting down, both of which seem to be solved by `HTTPClient.shared`.

If we don't go this route, then we probably still can get rid of this assuming we don't need to call `syncShutdown` ourselves, which it seems like we don't need to?

Reply:
>If we go the shared route, I don't think we need this class at all actually. The original intent was to be able to share a single HTTP client via class's arc semantics and also to ensure it shut down when the application was shutting down, both of which seem to be solved by HTTPClient.shared.

Right, this doesn't really work because you can't know what thread will call `deinit`, so if you call `syncShutdown` in there that can be pretty bad.

There are really two 'correct' ways:
1. in `main` call: `let client = HTTPClient(...); defer { try! client.syncShutdown() }` and dependency inject it
2. use a singleton. Either `HTTPClient.shared` or your own singleton 

Most server people prefer (1) but Apple's SDKs are typically singleton-heavy and therefore the Swift community is often quite keen on simplicity which would be (2). Especially for a command-line utility like Swiftly, having a HTTPClient-singleton isn't really that bad.

Of course it's a serious design issue in HTTPClient that some of the configuration (like the proxy settings) can only be configured using the top-level object (`HTTPClient`) that you're really meant to share everywhere (https://github.com/swift-server/async-http-client/issues/392).

Reply:
Yeah, I think a singleton would be fine for swiftly, it's basically what we have already. Until `shared` can be configured to handle the proxy settings, it seems like we'll need to do it ourselves though. I think we can do that in a separate commit / PR or here if you'd like, either is fine with me.